sdk/search/Azure.Search.Documents/src/Indexes/Models/IndexingParametersConfiguration.cs

```diff
@@ -0,0 +1,277 @@
+        {
+            get => Get<bool?>(FirstLineContainsHeadersKey);
+            set => Set(FirstLineContainsHeadersKey, value);
+        }
+
+        /// <summary> For JSON arrays, given a structured or semi-structured document, you can specify a path to the array using this property. </summary>
+        public string DocumentRoot
+        {
+            get => Get<string>(DocumentRootKey);
+            set => Set(DocumentRootKey, value);
+        }
+
+        /// <summary> Specifies the data to extract from Azure blob storage and tells the indexer which data to extract from image content when &quot;imageAction&quot; is set to a value other than &quot;none&quot;.  This applies to embedded image content in a .PDF or other application, or image files such as .jpg and .png, in Azure blobs. </summary>
+        public BlobIndexerDataToExtract? DataToExtract
+        {
+            get => Get(DataToExtractKey, value => new BlobIndexerDataToExtract(value));
+            set => Set(DataToExtractKey, value);
+        }
+
+        /// <summary> Determines how to process embedded images and image files in Azure blob storage.  Setting the &quot;imageAction&quot; configuration to any value other than &quot;none&quot; requires that a skillset also be attached to that indexer. </summary>
+        public BlobIndexerImageAction? ImageAction
+        {
+            get => Get(ImageActionKey, value => new BlobIndexerImageAction(value));
+            set => Set(ImageActionKey, value);
+        }
+
+        /// <summary> If true, will create a path //document//file_data that is an object representing the original file data downloaded from your blob data source.  This allows you to pass the original file data to a custom skill for processing within the enrichment pipeline, or to the Document Extraction skill. </summary>
+        public bool? AllowSkillsetToReadFileData
+        {
+            get => Get<bool?>(AllowSkillsetToReadFileDataKey);
+            set => Set(AllowSkillsetToReadFileDataKey, value);
+        }
+
+        /// <summary> Determines algorithm for text extraction from PDF files in Azure blob storage. </summary>
+        public BlobIndexerPdfTextRotationAlgorithm? PdfTextRotationAlgorithm
+        {
+            get => Get(PdfTextRotationAlgorithmKey, value => new BlobIndexerPdfTextRotationAlgorithm(value));
+            set => Set(PdfTextRotationAlgorithmKey, value);
+        }
+
+        /// <summary> Specifies the environment in which the indexer should execute. </summary>
+        public IndexerExecutionEnvironment? ExecutionEnvironment
+        {
+            get => Get(ExecutionEnvironmentKey, value => new IndexerExecutionEnvironment(value));
+            set => Set(ExecutionEnvironmentKey, value);
+        }
+
+        /// <summary> Increases the timeout beyond the 5-minute default for Azure SQL database data sources, specified in the format &quot;hh:mm:ss&quot;. </summary>
+        public string QueryTimeout
```

Comment:
From swagger

Reply:
Can you even have a TimeSpan in the spec or is this a .NET-ism? Since this is not generated code, I thought we could be more idiomatic than the spec can be.

Reply:
It's actually from generated code, but I might be able to declare it as a TimeSpan, or at least have a private field and make this a TimeSpan. I agree that would be better in this case. For string variants, while I appreciate what the old extension methods did, validating the format goes against our guidelines since it can change. But in this case, I think it makes sense.

Please be sure to unresolve any comments you want me to follow-up on. This was still resolved, but I just happened to notice it. I resolve comments as I deal with them so that unresolved make a sort of "todo list".